Model/SourcesRepository.php

```diff
@@ -0,0 +1,154 @@
+namespace Ampersand\DisableStockReservation\Model;
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Ampersand\DisableStockReservation\Model\SourcesFactory;
+use Ampersand\DisableStockReservation\Model\ResourceModel\Sources;
+use Ampersand\DisableStockReservation\Model\Sources as SourceModel;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterfaceFactory;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterface;
+use Ampersand\DisableStockReservation\Service\SourcesConverter;
+use Ampersand\DisableStockReservation\Model\ResourceModel\Sources\CollectionFactory;
+use Magento\Sales\Api\Data\OrderSearchResultInterface;
+use Ampersand\DisableStockReservation\Api\Data\SourcesInterface;
+
+/**
+ * Class SourcesRepository
+ * @package Ampersand\DisableStockReservation\Model
+ */
+class SourcesRepository
+{
+    /**
+     * @var SourcesFactory
+     */
+    protected $sourcesFactory;
+
+    /**
+     * @var Sources
+     */
+    protected $sourcesResourceModel;
+
+    /**
+     * @var SourceSelectionResultInterfaceFactory
+     */
+    private $sourceSelectionResultFactory;
+
+    /**
+     * @var SourcesConverter
+     */
+    private $sourcesConverter;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $collectionFactory;
+
+    /**
+     * SourcesRepository constructor.
+     * @param SourcesFactory $sourcesFactory
+     * @param Sources $sourcesResourceModel
+     * @param SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory
+     * @param SourcesConverter $sourcesConverter
+     * @param CollectionFactory $collectionFactory
+     */
+    public function __construct(
+        SourcesFactory $sourcesFactory,
+        Sources $sourcesResourceModel,
+        SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory,
+        SourcesConverter $sourcesConverter,
+        CollectionFactory $collectionFactory
+    ) {
+        $this->sourcesFactory = $sourcesFactory;
+        $this->sourcesResourceModel = $sourcesResourceModel;
+        $this->sourceSelectionResultFactory = $sourceSelectionResultFactory;
+        $this->sourcesConverter = $sourcesConverter;
+        $this->collectionFactory = $collectionFactory;
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceModel
+     * @throws \Exception
+     */
+    public function getByOrderId(string $orderId): SourceModel
+    {
+        /** @var SourceModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+        $this->sourcesResourceModel->load(
+            $sourcesModel,
+            $orderId,
+            'order_id'
+        );
+
+        if (!$sourcesModel->getId()) {
+            throw new NoSuchEntityException(__('Source model with the order ID "%1', $orderId));
+        }
+
+        return $sourcesModel;
+    }
+
+    public function getOrdersSourcesCollection(OrderSearchResultInterface $result)
```

Comment:
This is looking much better, a couple of things to tidy up:
- Missing docblocks
- Missing return type
- `$resultIds []` should be `$resultIds[]`
- This is no longer returning a collection of objects, maybe we should update the functions name to reflect this?
```php
$orderSources[$item->getOrderId()] = $this->sourcesConverter->
 convertSourcesArrayToSourceSelectionItems($item->getSources());
```
If the function needs to span 2 lines we'd normally do something like (at least in PSR-2):
```php
$orderSources[$item->getOrderId()] = $this->sourcesConverter
    ->convertSourcesArrayToSourceSelectionItems($item->getSources());
```
Or:
```php
$orderSources[$item->getOrderId()] = $this->sourcesConverter->convertSourcesArrayToSourceSelectionItems(
    $item->getSources()
);
```
Or even:
```php
$orderSources[$item->getOrderId()] =
    $this->sourcesConverter->convertSourcesArrayToSourceSelectionItems($item->getSources());
```
However I don't think this should reside in the repository... The repository should ideally be only responsible for the CRUD actions associated with our model. This functionality should really be in the plugin where it's the only use case